lib/puppetserver/ca/action/generate.rb

```diff
@@ -202,13 +202,22 @@ def generate_certs(certnames, alt_names, settings, digest)
             key, csr = generate_key_csr(certname, settings, digest, current_alt_names)
             next false unless csr
             next false unless ca.submit_certificate_request(certname, csr)
-            next false unless ca.sign_certs([certname])
+
+            # Check if the CA autosigned the cert
```

Comment:
These bits inside the `.map` feel like they should be a separate method or three? I'm not sure what the best way to break them up would be though.